### PR TITLE
Implement NavList component (part 1)

### PR DESCRIPTION
**Pull request by colebemis**
Implements the NavList component as documented in [this PR](https://github.com/primer/react/pull/2027).

✨ [Docs preview](https://primer-f1dedbce87-13348165.drafts.github.io/NavList) ✨ 

### Screenshots

CleanShot 2022-05-05 at 16 18 58@2x

## A11y review notes

NavList [groups](https://primer-f1dedbce87-13348165.drafts.github.io/NavList#with-groups) use the ActionList.Group component directly but render markup that's different than the recommended markup in [this a11y review](https://github.com/github/primer/issues/651). Should groups be rendered differently in NavList vs ActionList? cc @alliethu 

## Code review notes

In an effort to keep PRs small and easy to review, I plan to implement NavList incrementally. In future PRs, I plan to implement:
- [ ] Sub-item support
- [ ] Support Next.js and React Router links via the `as` prop
- [ ] Support `ref` prop on all subcomponents
- [ ] Support for sx prop (with deepmerge)

I'll leave `NavList` in the `@primer/react/drafts` bundle until it is feature-complete and meets all the criteria of an [alpha](https://primer.style/contribute/component-lifecycle#alpha) component. 

## Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

---

Part of https://github.com/github/primer/issues/637

### CODE REVIEW

File: src/NavList/NavList.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  children: React.ReactNode
+  'aria-label'?: string
+  'aria-labelledby'?: string
+  // sx
+}
+
+const Root = React.forwardRef<HTMLElement, NavListProps>(
+  ({'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledBy, children}, ref) => {
+    return (
+      <nav ref={ref} aria-label={ariaLabel} aria-labelledby={ariaLabelledBy}>
```
Comment (siddharthkp) [inline src/NavList/NavList.tsx:17]: Should we pass down all `...props` for things like `id` / `data-test` etc?

File: docs/content/NavList.mdx [hunk 2]
```diff
@@ -154,31 +86,13 @@ To render a horizontal list of navigation links, consider using [UnderlineNav](/
...
-
-</details>
-
 ### With a heading
 
-```jsx
+```jsx live drafts
 <>
-  <h3 id="workflows-heading">Workflows</h3>
+  <Heading as="h3" id="workflows-heading" sx={{fontSize: 2}}>
```
Comment (siddharthkp) [inline docs/content/NavList.mdx:93]: Good example!

File: docs/content/NavList.mdx [hunk 3]
```diff
@@ -325,9 +183,11 @@ If a `NavList.Item` contains a `NavList.SubNav`, the `NavList.Item` will render
...
-```jsx
+```jsx live drafts
 <NavList>
-  <NavList.Item href="/">Dashboard</NavList.Item>
+  <NavList.Item href="/" aria-current="page">
+    Dashboard
+  </NavList.Item>
   <NavList.Item href="/pulls">Pull requests</NavList.Item>
   <NavList.Item href="/issues">Issues</NavList.Item>
   <NavList.Divider />
```
Comment (siddharthkp) [inline docs/content/NavList.mdx:193]: This is interesting 🤔 

Would we also want to support `<NavList showDivider={true}` [similar to Action list](https://primer.style/react/ActionList#:~:text=can%20be%20selected.-,showDivider,-boolean)?

File: src/NavList/NavList.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+const Item = React.forwardRef<HTMLAnchorElement, NavListItemProps>(
+  ({href, 'aria-current': ariaCurrent, children}, ref) => {
+    return (
+      <ActionList.LinkItem
+        ref={ref}
+        href={href}
+        aria-current={ariaCurrent}
+        sx={{
+          position: 'relative',
+          '&[aria-current]': {
```
Comment (siddharthkp) [inline src/NavList/NavList.tsx:45]: should this be in ActionList? other components like `AutoComplete` and `SelectPanel` would reuse the styles for active descendent

File: src/NavList/NavList.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        }}
+      >
+        {children}
+      </ActionList.LinkItem>
+    )
+  }
+)
+
+// ----------------------------------------------------------------------------
+// NavList.LeadingVisual
```
Comment (siddharthkp) [inline src/NavList/NavList.tsx:68]: Pretty comment aesthetics!

**Review by siddharthkp — APPROVED** [review]
Looks really good! I think we can already start using it in doctocat

Left a few questions and suggestions :)

**Comment by siddharthkp** [comment]
For the checklist:
- support for sx prop (with deepmerge)
- do you have a use case for https://github.com/primer/react/issues/2043?

File: docs/content/NavList.mdx (hunk 3 above)
Comment (colebemis) [reply docs/content/NavList.mdx:193]: I don't think dividers in between individual items is part of the NavList design cc @vdepizzol

File: src/NavList/NavList.tsx (hunk 1 above)
Comment (colebemis) [reply src/NavList/NavList.tsx:17]: Sure, I'll pass down all props accepted by the `<nav>` element 👍

File: src/NavList/NavList.tsx (hunk 4 above)
Comment (colebemis) [reply src/NavList/NavList.tsx:45]: Yeah, I think that makes sense. Gonna leave it here for now, then abstract it in a separate PR.

**Comment by colebemis** [comment]
> do you have a use case for https://github.com/primer/react/issues/2043?

Yes I think so!

CleanShot 2022-05-09 at 14 27 07@2x

File: src/NavList/NavList.tsx (hunk 1 above)
Comment (colebemis) [reply src/NavList/NavList.tsx]: Done ✅

**Pull request merged**